modules/tinymce/src/core/test/ts/atomic/html/FilterNodeTest.ts

```diff
@@ -0,0 +1,47 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { assert } from 'chai';
+
+import { AstNode } from 'tinymce/core/api/PublicApi';
+import { traverse } from 'tinymce/core/html/FilterNode';
+
+describe('browser.tinymce.core.html.FilterNode', () => {
```

Comment:
```suggestion
describe('atomic.tinymce.core.html.FilterNode', () => {
```

Reply:
Also we don't appear to have a test that verifies this hasn't regressed... can we please add one for that? We should be able to do that something like this (or at least this fails using the old code for me testing locally):
```ts
const body = new AstNode('body', 11);
Arr.range(15000, () => {
	body.append(new AstNode('p', 1));
});

traverse(body, Fun.noop);
```

- [ ] Add test to prevent recursion error

Reply:
fixed and test added